perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -660,6 +673,26 @@ sub map :Chained('id') :Args(0) {
     $c->res->body($image->{data});
 }
 
+sub confirmation :Chained('id') :Args(0) {
+    my ($self, $c) = @_;
+
+    my $template = $c->get_param('template') || "";
+
+    if ($template !~ /^(email_sent|confirm_problem)$/) {
+        $c->detach( '/page_error_404_not_found', [] );
```

Comment:
This includes the report's location on the 404 page if you go to /report/ID/confirmation for any ID on any cobrand, would probably be better if it didn't, even if the report isn't non_public.

Reply:
There's no longer a report on the stash at this point so shouldn't happen any more.